[ANGLE] Update ANGLE to 2024-09-10 (941b3df3df32a06a4b8a01c41c5522e2ea859d04)

djg opened the pull request:
#### 10807663197aa42f1df3eaf7fd68b3e4b18df197
<pre>
[ANGLE] Update ANGLE to 2024-09-10 (941b3df3df32a06a4b8a01c41c5522e2ea859d04)
<a href="https://bugs.webkit.org/show_bug.cgi?id=279413">https://bugs.webkit.org/show_bug.cgi?id=279413</a>
<a href="https://rdar.apple.com/135645407">rdar://135645407</a>

Reviewed by Kimmo Kinnunen

Contains upstream commits:
git log --oneline 5a8eab96c6b7b99b8568feb902b8b85e2df7e19a..941b3df3df32a06a4b8a01c41c5522e2ea859d04 --pretty=%h %s
941b3df3df Vulkan: Add new threshold for CPU buffer subdata
53476d6ff2 Vulkan: Do vertex conversion with fine grain dirtyRange
e3e59b24ce ssci: mark StackWalker as &quot;Version: N/A&quot;
a0126d5515 Metal: Clear mRenderPassesSinceFlush to avoid infinite recursion
bda335985b Test: Add Critical Ops trace
22c39d24d2 Android aosp: remove sdk_version from most rules (try 2)
bfadbe808c Revert &quot;Tests: temporarily disable end2end test .so library in AOSP&quot;
18aeee88fe Android aosp: remove sdk_version from most rules
67b3149f2c Roll VK-GL-CTS from 7d2299e67fe7 to c3450b232f2c (8 revisions)
681a3cacc8 Roll Chromium from 1640f4fb3586 to 55810c9b176d (683 revisions)
fc65058c15 CL/Vulkan: Rework spec constant handling
f063960c9e roll_aosp.sh: keep third_party/jdk/BUILD.gn
e19377af63 Tests: Support multiple users on Android
2deb9534d5 Tests: temporarily disable end2end test .so library in AOSP
45ea20944a Don&apos;t build api_dump for secondary abi on Android
20511d0bb7 Vulkan: Remove unnecessary perfetto trace in BufferVk::mapRange
49a8d05c90 Tests: Add end2end test .so library to AOSP build
887d148a7e Roll Chromium from 93076992192b to 1640f4fb3586 (640 revisions)
61c26fd693 Tests: Add Passmark benchmark Unity trace
0a04cb905f Translator: Limit shader size check to WebGL and Chromium shaders
34201e0d44 Roll Chromium from e8f446cd167a to 93076992192b (563 revisions)
59eff3660f CL/Vulkan: Enable global offset support
621bba6cc6 Vulkan: Fix device entry points loaded as instance
ec8879a2f6 Roll Chromium from def24c080915 to e8f446cd167a (673 revisions)
f0d0b01475 Roll vulkan-deps from aa237b66d8cc to 725499142cb6 (6 revisions)
2d9a2bcb6a Roll SwiftShader from 3c4bdf66d81d to 5561c71fa64e (1 revision)
d1a4b0ff5b Remove Feature Level 9_3 code
e13cc7df70 Metal: Add shadow sampler overloads for textureProj[Offset]
177d15b380 Vulkan: Simplify WindowSurfaceVk::prepareSwap() method
edc6d2fc4a Metal: Fix watchOS supports32BitFloatFiltering use
8b8b3d4d3a Metal: Flush with NoWait on iOS when hitting renderpass count limits
e5532e6c7f Metal: Avoid using Display::prepareForCall
c997a32bff Roll vulkan-deps from c9475650ea41 to aa237b66d8cc (14 revisions)
5108112e4c Roll Chromium from 88be4af3fd72 to def24c080915 (266 revisions)
cf9e7fa532 Roll vulkan-deps from 183438492d3f to c9475650ea41 (2 revisions)
377f7b68e8 Vulkan: Remove redundant needs-flush check in glClientWaitSync
ac82ae706f Roll VK-GL-CTS from f1cd1d841346 to 7d2299e67fe7 (6 revisions)
636d838e66 Vulkan: Decouple Window Surface width/height from color RT
74816e9e08 Vulkan: Cleanup RenderTargetVk release/destroy methods
5d306ca4eb Roll Chromium from 32ac9da2c5a2 to 88be4af3fd72 (439 revisions)
8b6b1113b9 Vulkan: Fallback to host unCached if no cached memory available
21b286a19d Trace replay: Call glFinish before screenshots
20e0b83ac4 Vulkan: Expand preferSubmitAtFBOBoundary from ARM to all tiler GPUs
56ea2edc02 Traces: Update regex for power rails
a2a9c686bf Manual roll Chromium from b5ec9af34e67 to 32ac9da2c5a2 (781 revisions)
e4be31cafc Test: Add VertexAttribPointer test with different offset
4167a9a05a Don&apos;t lock the context mutex if no flush in eglClientWaitSync
4e3b4e0c2d Vulkan: Clear conversion buffer&apos;s dirtyRange if bufferData called
27a0a5edab Vulkan: Only access ContextVk in EGLSync if asked to flush
e4aab5ccc2 Vulkan: Reuse the same conversion buffer by adjusting offset
4dfba10af7 Skip passmark_complex on Linux Intel
8d6f6c23e5 Vulkan: Unsuppress fixed VVL errors
d886a0fbcb Metal: Remove unused platform macros
2eb9b102f6 Roll vulkan-deps from e5360151769c to 183438492d3f (6 revisions)
8c211b10e5 Metal: Clear depth pbuffers to 1 for robust init
e2afdd3ca3 Metal: Fix depth/stencil FBO attachments validation
b43f888968 Vulkan: Fix varyings reserved for xfb
28d82804f7 Tests: Add Passmark benchmark Complex trace
e850656a4a Add EXT_separate_depth_stencil stubs
eba5acd1a8 Roll Chromium from d0115bcb53a9 to b5ec9af34e67 (674 revisions)
c3309dbd3b Tests: Add Passmark benchmark Simple trace
fffd48d302 Roll vulkan-deps from 23ed8d76c58a to e5360151769c (7 revisions)
ca20a67c3a Roll VK-GL-CTS from 824d14748364 to f1cd1d841346 (1 revision)
79729f7c60 Reapply &quot;Vulkan: Expose float control properties as angle features&quot;
c094d4bc23 Enable tests on linux-exp-nvidia
0929a8d720 CL/Vulkan: Add utility function for clspv compile options
42ea855519 GL: Sync default VAO attributes for external contexts.
8522e83594 WebGPU: Avoid poping error scopes too many times
24a3d30d02 Possibly fix FixedQueue.ConcurrentPushPopWithResize flakiness
f25948774f Roll vulkan-deps from 7efe15073686 to 23ed8d76c58a (4 revisions)
23998c40f0 Roll Chromium from 4358bf5ee791 to d0115bcb53a9 (374 revisions)
cd4b8adb59 Remove obsolete macOS availability macros
3f5ede8f11 Metal: Remove obsolete availability macros from context
e39d08b9a2 Metal: Remove obsolete code from caps and extensions init
851f949bc3 Manual roll Chromium from 28739f6b546a to 4358bf5ee791 (141 revisions)
9e8d632a50 Guard against find_bad_constructs on nacl (again)
fc4f0ac0df formatutils: Allow 16-bit formats for renderbuffer with ext in GLES1.0
e9282ae72e Manual roll Chromium from eaaa9b6d35f9 to 28739f6b546a (100 revisions)
b512e42d25 Reland: Metal: Fix availability for kIOMainPortDefault
b52f281a95 Metal: Remove unused and obsolete availability macros
2609e02c9a Metal: Remove unused branches from StartFrameCapture
8cfb9ea255 Metal: Align OS macro usage
be4532beb2 Metal: Update availability macros in tests
ecdc98f4ea Roll Chromium from eca761dbc106 to eaaa9b6d35f9 (664 revisions)
7108f0b40a Roll vulkan-deps from 38174d6926f6 to 7efe15073686 (6 revisions)
9bf6fb58c3 Roll SwiftShader from dc27069e5d04 to 3c4bdf66d81d (1 revision)

Canonical link: <a href="https://commits.webkit.org/283468@main">https://commits.webkit.org/283468@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/b3615c9d9d22bafce712c0fb3ec0a78acc475e29

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/66384 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/45759 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/19005 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/70417 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/59/builds/16995 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/53558 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/17277 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/53242 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 wincairo-tests](https://ews-build.webkit.org/#/builders/60/builds/11855 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/69451 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/42184 "Passed tests") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/18/builds/57467 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-wpe~~](https://ews-build.webkit.org/#/builders/41/builds/33896 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/38855 "Passed tests") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/64/builds/14855 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/15870 "Built successfully") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/60750 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/63/builds/15197 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/72120 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/87/builds/10341 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/62/builds/14570 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/60566 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/86/builds/10373 "Built successfully") | [  ~~🧪 mac-wk2-stress~~](https://ews-build.webkit.org/#/builders/8/builds/57536 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/21/builds/60880 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/14625 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/88/builds/8538 "Passed tests") | [  ~~🧪 mac-intel-wk2~~](https://ews-build.webkit.org/#/builders/119/builds/2148 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [  ~~🛠 tv~~](https://ews-build.webkit.org/#/builders/44/builds/41566 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/42643 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/43826 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/42386 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

kkinnunen-apple commented on Source/ThirdParty/ANGLE/changes.diff:
```diff
@@ -1368,6 +1368,140 @@ index ec84d044755d61e14353fb32735a7d13a60d19dd..c3f918bb49ca3d10bcd767079e8c6f7e
    protected:
      const gl::State &mState;
      gl::MemoryProgramCache *mMemoryProgramCache;
+diff --git a/src/libANGLE/renderer/d3d/d3d11/shaders/compiled/clearfloat11_fl9ps.h b/src/libANGLE/renderer/d3d/d3d11/shaders/compiled/clearfloat11_fl9ps.h
```
Something strange here with the file?

kkinnunen-apple reviewed (approved):
With the diff + the missing file (??) fixed

djg committed:
[ANGLE] Update ANGLE to 2024-09-10 (941b3df3df32a06a4b8a01c41c5522e2ea859d04)
https://bugs.webkit.org/show_bug.cgi?id=279413
rdar://135645407

Reviewed by Kimmo Kinnunen

Contains upstream commits:
git log --oneline 5a8eab96c6b7b99b8568feb902b8b85e2df7e19a..941b3df3df32a06a4b8a01c41c5522e2ea859d04 --pretty=%h %s
941b3df3df Vulkan: Add new threshold for CPU buffer subdata
53476d6ff2 Vulkan: Do vertex conversion with fine grain dirtyRange
e3e59b24ce ssci: mark StackWalker as "Version: N/A"
a0126d5515 Metal: Clear mRenderPassesSinceFlush to avoid infinite recursion
bda335985b Test: Add Critical Ops trace
22c39d24d2 Android aosp: remove sdk_version from most rules (try 2)
bfadbe808c Revert "Tests: temporarily disable end2end test .so library in AOSP"
18aeee88fe Android aosp: remove sdk_version from most rules
67b3149f2c Roll VK-GL-CTS from 7d2299e67fe7 to c3450b232f2c (8 revisions)
681a3cacc8 Roll Chromium from 1640f4fb3586 to 55810c9b176d (683 revisions)
fc65058c15 CL/Vulkan: Rework spec constant handling
f063960c9e roll_aosp.sh: keep third_party/jdk/BUILD.gn
e19377af63 Tests: Support multiple users on Android
2deb9534d5 Tests: temporarily disable end2end test .so library in AOSP
45ea20944a Don't build api_dump for secondary abi on Android
20511d0bb7 Vulkan: Remove unnecessary perfetto trace in BufferVk::mapRange
49a8d05c90 Tests: Add end2end test .so library to AOSP build
887d148a7e Roll Chromium from 93076992192b to 1640f4fb3586 (640 revisions)
61c26fd693 Tests: Add Passmark benchmark Unity trace
0a04cb905f Translator: Limit shader size check to WebGL and Chromium shaders
34201e0d44 Roll Chromium from e8f446cd167a to 93076992192b (563 revisions)
59eff3660f CL/Vulkan: Enable global offset support
621bba6cc6 Vulkan: Fix device entry points loaded as instance
ec8879a2f6 Roll Chromium from def24c080915 to e8f446cd167a (673 revisions)
f0d0b01475 Roll vulkan-deps from aa237b66d8cc to 725499142cb6 (6 revisions)
2d9a2bcb6a Roll SwiftShader from 3c4bdf66d81d to 5561c71fa64e (1 revision)
d1a4b0ff5b Remove Feature Level 9_3 code
e13cc7df70 Metal: Add shadow sampler overloads for textureProj[Offset]
177d15b380 Vulkan: Simplify WindowSurfaceVk::prepareSwap() method
edc6d2fc4a Metal: Fix watchOS supports32BitFloatFiltering use
8b8b3d4d3a Metal: Flush with NoWait on iOS when hitting renderpass count limits
e5532e6c7f Metal: Avoid using Display::prepareForCall
c997a32bff Roll vulkan-deps from c9475650ea41 to aa237b66d8cc (14 revisions)
5108112e4c Roll Chromium from 88be4af3fd72 to def24c080915 (266 revisions)
cf9e7fa532 Roll vulkan-deps from 183438492d3f to c9475650ea41 (2 revisions)
377f7b68e8 Vulkan: Remove redundant needs-flush check in glClientWaitSync
ac82ae706f Roll VK-GL-CTS from f1cd1d841346 to 7d2299e67fe7 (6 revisions)
636d838e66 Vulkan: Decouple Window Surface width/height from color RT
74816e9e08 Vulkan: Cleanup RenderTargetVk release/destroy methods
5d306ca4eb Roll Chromium from 32ac9da2c5a2 to 88be4af3fd72 (439 revisions)
8b6b1113b9 Vulkan: Fallback to host unCached if no cached memory available
21b286a19d Trace replay: Call glFinish before screenshots
20e0b83ac4 Vulkan: Expand preferSubmitAtFBOBoundary from ARM to all tiler GPUs
56ea2edc02 Traces: Update regex for power rails
a2a9c686bf Manual roll Chromium from b5ec9af34e67 to 32ac9da2c5a2 (781 revisions)
e4be31cafc Test: Add VertexAttribPointer test with different offset
4167a9a05a Don't lock the context mutex if no flush in eglClientWaitSync
4e3b4e0c2d Vulkan: Clear conversion buffer's dirtyRange if bufferData called
27a0a5edab Vulkan: Only access ContextVk in EGLSync if asked to flush
e4aab5ccc2 Vulkan: Reuse the same conversion buffer by adjusting offset
4dfba10af7 Skip passmark_complex on Linux Intel
8d6f6c23e5 Vulkan: Unsuppress fixed VVL errors
d886a0fbcb Metal: Remove unused platform macros
2eb9b102f6 Roll vulkan-deps from e5360151769c to 183438492d3f (6 revisions)
8c211b10e5 Metal: Clear depth pbuffers to 1 for robust init
e2afdd3ca3 Metal: Fix depth/stencil FBO attachments validation
b43f888968 Vulkan: Fix varyings reserved for xfb
28d82804f7 Tests: Add Passmark benchmark Complex trace
e850656a4a Add EXT_separate_depth_stencil stubs
eba5acd1a8 Roll Chromium from d0115bcb53a9 to b5ec9af34e67 (674 revisions)
c3309dbd3b Tests: Add Passmark benchmark Simple trace
fffd48d302 Roll vulkan-deps from 23ed8d76c58a to e5360151769c (7 revisions)
ca20a67c3a Roll VK-GL-CTS from 824d14748364 to f1cd1d841346 (1 revision)
79729f7c60 Reapply "Vulkan: Expose float control properties as angle features"
c094d4bc23 Enable tests on linux-exp-nvidia
0929a8d720 CL/Vulkan: Add utility function for clspv compile options
42ea855519 GL: Sync default VAO attributes for external contexts.
8522e83594 WebGPU: Avoid poping error scopes too many times
24a3d30d02 Possibly fix FixedQueue.ConcurrentPushPopWithResize flakiness
f25948774f Roll vulkan-deps from 7efe15073686 to 23ed8d76c58a (4 revisions)
23998c40f0 Roll Chromium from 4358bf5ee791 to d0115bcb53a9 (374 revisions)
cd4b8adb59 Remove obsolete macOS availability macros
3f5ede8f11 Metal: Remove obsolete availability macros from context
e39d08b9a2 Metal: Remove obsolete code from caps and extensions init
851f949bc3 Manual roll Chromium from 28739f6b546a to 4358bf5ee791 (141 revisions)
9e8d632a50 Guard against find_bad_constructs on nacl (again)
fc4f0ac0df formatutils: Allow 16-bit formats for renderbuffer with ext in GLES1.0
e9282ae72e Manual roll Chromium from eaaa9b6d35f9 to 28739f6b546a (100 revisions)
b512e42d25 Reland: Metal: Fix availability for kIOMainPortDefault
b52f281a95 Metal: Remove unused and obsolete availability macros
2609e02c9a Metal: Remove unused branches from StartFrameCapture
8cfb9ea255 Metal: Align OS macro usage
be4532beb2 Metal: Update availability macros in tests
ecdc98f4ea Roll Chromium from eca761dbc106 to eaaa9b6d35f9 (664 revisions)
7108f0b40a Roll vulkan-deps from 38174d6926f6 to 7efe15073686 (6 revisions)
9bf6fb58c3 Roll SwiftShader from dc27069e5d04 to 3c4bdf66d81d (1 revision)

Canonical link: https://commits.webkit.org/283468@main
